File: 69. Sqrt(x)/69. Sqrt(x).java

class Solution {

    public int mySqrt(int x) {
        if (x < 2) {
            return x;
        }
        long l = 1;
        long r = x / 2;
        while (l <= r) {
            long m = (l + r) / 2;
            if (m * m == x) {
                return (int) Math.floor(m);
            } else if (m * m < x) {
                l = m + 1;
            } else {
                r = m - 1;
            }
        }
        return (int) Math.floor(r);
    }
}

File: 69. Sqrt(x)/69. Sqrt(x).py
import math

class Solution:
    def mySqrt(self, x: int) -> int:
        if x < 2:
            return math.floor(x)
        left, right = 1, x//2
        while left <= right:
            mid = (left + right) // 2
            if mid * mid == x:
                return math.floor(mid)
            elif mid * mid < x:
                left = mid + 1
            else:
                right = mid - 1
        return math.floor(right)
                